app/src/ereferrals/java/org/eyeseetea/malariacare/data/sync/importer/strategies/PullControllerStrategy.java

```diff
@@ -73,12 +76,20 @@ public void onPullDataComplete(final IPullController.Callback callback, boolean
                         orgUnitDataSource.getUserOrgUnit(
                                 credentialsLocalDataSource.getOrganisationCredentials());
                 org.eyeseetea.malariacare.domain.entity.Program program = orgUnit.getProgram();
-                programLocalDataSource.saveUserProgramId(program);
+                saveProgram(programLocalDataSource, program);
             } catch (Exception e) {
                 e.printStackTrace();
                 callback.onError(e);
             }
             mPullController.convertData(callback);
         }
     }
+
+    private void saveProgram(IProgramRepository programLocalDataSource, Program program) {
+        if(!programLocalDataSource.checkLastDownloadedProgramMedia(program.getCode())){
```

Comment:
sorry ok is true, I think is better save in every mediaDB the program Id, It is more natural because you can to use MediaRepository for save this info and the creation of a new repository is not necessary. On this code in pull process, you can retrieve first media from MediaRepository to know program uid used on last download media process.